mypy/newsemanal/semanal_main.py

```diff
@@ -54,8 +63,10 @@ def process_functions(graph: 'Graph', scc: List[str]) -> None:
         symtable = tree.names
         targets = get_all_leaf_targets(symtable, module, None)
         for target, node, active_type in targets:
-            deferred += semantic_analyze_target(target, graph[module], node, active_type)
-    assert not deferred  # There can't be cross-function forward refs
+            deferred, incomplete = semantic_analyze_target(target, graph[module], node,
+                                                           active_type)
+            assert not deferred  # There can't be cross-function forward refs
+            assert not incomplete  # Ditto
```

Comment:
Ah I didn't consider this, TBH. Since the namespaces of functions don't survive semantic analysis, I think that this should be handled internally within the semantic analyzer, which means that the deferrals wouldn't be exposed to `semanal_main`. We'd perform multiple iterations in `refresh_partial` and only return after we've bound all names.